### PR TITLE
School library Association

### PR DESCRIPTION
**Pull request by ridabensalem**
in this milestone of the School library project, we have added the following requirements : 
- Create a class Classroom with the following:
1. @label instance variable, initialized in the constructor.
2. Setter and getter for @label .
3. Create the has-many/belongs-to relationship between Classroom and Student. 
4. Create the has-many side (a classroom has many students).
5. Create the belongs-to side (a student belongs to a classroom).
- Create a class Book with the following:
1. @title and @author instance variables.
2. Setters and getters for instance variables.
- Create a class Rental with the following:
1.  @date instance variable  initialized in the constructor.
2. Setter and getter for @date .
3. Create the many-to-many (also has-many-through) relationship between Person and Book using the intermediate class Rental.
4. Create the has-many side of Book and Rental (a book has many rentals).
5. Create the belongs-to side of Rental and Book (a rental belongs to a book).
6. Create the has-many side of Person and Rental (a person has many rentals).
7. Create the belongs-to side of Rental and Person (a rental belongs to a person).
8. Modify the constructor of Rental so Book and Person are set in it.

### CODE REVIEW

File: classroom.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+  def students
+    Student.all.select do |student|
+      student.classroom == self
+    end
+  end
+
+  def add_student(student)
+    student.classroom = self
+  end
```
Comment (Olamarx) [inline classroom.rb]: - [ ] You need to push the student to the student variable you initialized too. Something like: `@student << student`

File: classroom.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class Classroom
+  attr_accessor :label
+
```
Comment (Olamarx) [inline classroom.rb]: - [ ] You need to make the student readable/getter by setting the getter `attr_reader :student`

File: book.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+require_relative 'rental'
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rentals = []
+  end
```
Comment (Olamarx) [inline book.rb:9]: - [ ] In this file, an `add_rental` method too needs to be added as required and inside it, the book itself should be sent as a parameter to create the new rental achieving the 'has-many' association.

**Review by Olamarx — CHANGES_REQUESTED** [review]
Hi @ridabensalem, 

Good job so far, indeed you have spent time on this project!
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- Linters passed ✔️
- Commit messages ✔️
- Gitflow ✔️

## Required Changes ♻️
_Check the comments under the review._
- [ ] You need to also create a method that serves the belong_to side of it in the Student class. It can take this form:
```
def classroom=(classroom)
      @classroom = classroom
      classroom.students.push(self) unless classroom.students.include?(self)
  end
```
- [ ] In the person.rb, getter and setter for rentals should be present and also initialized as a variable with an empty array. An add_rental method is also required so that the person itself should be sent as a parameter to create the new rental achieving the 'has-many' association.
## Optional suggestions
- [OPTIONAL] I noticed your readme file shows some raw code which is as a result of none proper closure with backtick, kindly put the closing three backticks to the next line.
image
 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: book.rb (hunk 3 above)
Comment (Davidosky007) [reply book.rb:9]: * As mentioned by the previous CR this issue is still pending Kindly fix it [refer to this lesson link](
https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship)

File: classroom.rb (hunk 2 above)
Comment (Davidosky007) [reply classroom.rb]: * This issue has not been fixed yet, kindly look into it

File: classroom.rb (hunk 1 above)
Comment (Davidosky007) [reply classroom.rb]: * As mentioned by the previous CR , Kindly work on this HINT 💭 see the example below 👇 [visit this link for more ](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships)
```
class Owner
  ...

  def add_animal(animal)
    @animals.push(animal)
    animal.owner = self
  end
end
```

**Review by Davidosky007 — CHANGES_REQUESTED** [review]
Hi @ridabensalem 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
* Well written PR 👍 
* Associations between Classes and Objects setup 👍 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @davidosky007 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by prolajumokeoni — APPROVED** [review]
Hi @ridabensalem 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
## Highlights
- All changes requested have been implemented 🥇  

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification @prolajumokeoni.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**